kanidm_client/tests/common.rs

```diff
@@ -25,7 +33,12 @@ pub fn run_test(test_fn: fn(KanidmClient) -> ()) {
     let (mut ready_tx, mut ready_rx) = mpsc::channel(1);
     let (mut finish_tx, mut finish_rx) = mpsc::channel(1);
 
-    let port = PORT_ALLOC.fetch_add(1, Ordering::SeqCst);
+    let port = loop {
```

Comment:
I think we should bound this loop to a set number of attempts so that we don't wait forever, maybe 5 or 10 attempts?

Reply:
Ok, add it because I don't see clearly how to do it without adding to much complexity for that non critical code.

Reply:
All good, I'll add the change myself :)